torch/cuda/__init__.py

```diff
@@ -267,6 +267,11 @@ def current_blas_handle():
     return torch._C._cuda_getCurrentBlasHandle()
 
 
+def empty_cache():
+    """Frees all unused memory currently held by the cache allocator"""
```

Comment:
I'm not sure how to phrase this, but maybe it's worth pointing out that not all of your free memory will be released. For example, if you:
1) allocate 10 GB
2) free 10 GB
3) allocate 2 MB
4) empty_cache()

You'll still have ~9.998 GB showing up as used in nvidia-smi because the 10GB segment is occupied by the 2 MB in-use allocation.

Reply:
Yes, that is why I had "unused" in there. But I'll update it to be clearer :)

Reply:
The way I'm reading caching allocator code, when 2 mb block is allocated, the remainder (it's called "remaining" in the code)  will be chopped of and marked as free, and then freed in empty_cache, so only 2 mb will show up in nvidia-smi.